File: src/test/java/org/trp/shincolle/block/VolBlockRegressionTest.java
package org.trp.shincolle.block;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

import static org.junit.jupiter.api.Assertions.assertTrue;

class VolBlockRegressionTest {

    private static final Path VOL_BLOCK_SOURCE =
            Path.of("src/main/java/org/trp/shincolle/block/VolBlock.kt");
    private static final Path MOD_BLOCKS_SOURCE =
            Path.of("src/main/java/org/trp/shincolle/init/ModBlocks.kt");
    private static final Path MOD_ITEMS_SOURCE =
            Path.of("src/main/java/org/trp/shincolle/init/ModItems.kt");
    private static final Path VOL_BLOCK_RECIPE =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolblock.json");
    private static final Path VOL_BLOCK_RECIPE_ALT =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolblock_2.json");
    private static final Path VOL_CORE_RECIPE =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolcore.json");
    private static final Path VOL_CORE_RECIPE_ALT =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolcore_3.json");

    @Test
    void volBlockShouldStayRestoredForLegacyVolcanoCraftChain() throws IOException {
        String blockSource = Files.readString(VOL_BLOCK_SOURCE);
        String modBlocksSource = Files.readString(MOD_BLOCKS_SOURCE);
        String modItemsSource = Files.readString(MOD_ITEMS_SOURCE);
        String volBlockRecipe = Files.readString(VOL_BLOCK_RECIPE);
        String volBlockRecipeAlt = Files.readString(VOL_BLOCK_RECIPE_ALT);
        String volCoreRecipe = Files.readString(VOL_CORE_RECIPE);
        String volCoreRecipeAlt = Files.readString(VOL_CORE_RECIPE_ALT);

        assertTrue(modBlocksSource.contains("VOL_BLOCK = BLOCKS.register(\"blockvolblock\""),
                "ModBlocks should keep registering the legacy volcano block");
        assertTrue(modItemsSource.contains("VOL_BLOCK = ITEMS.register(\"blockvolblock\""),
                "ModItems should keep exposing the volcano block as an item");
        assertTrue(blockSource.contains("return true;"),
                "VolBlock should stay valid as a beacon base like the legacy block");
        assertTrue(volBlockRecipe.contains("\"id\": \"shincolle:blockvolblock\""),
                "VolBlock recipe should stay present");
        assertTrue(volBlockRecipe.contains("\"item\": \"shincolle:grudge_block\""),
                "VolBlock recipe should keep using the legacy grudge block");
        assertTrue(volBlockRecipeAlt.contains("\"id\": \"shincolle:blockvolblock\""),
                "VolBlock should keep the mirrored legacy recipe variant");
        assertTrue(volCoreRecipe.contains("\"item\": \"shincolle:blockvolblock\""),
                "VolCore recipe should keep consuming the restored volcano block");
        assertTrue(volCoreRecipeAlt.contains("\"id\": \"shincolle:blockvolcore\""),
                "VolCore should keep the mirrored legacy recipe variant");
    }
}

File: src/test/kotlin/org/trp/shincolle/block/VolBlockRegressionTest.kt
package org.trp.shincolle.block

import org.junit.jupiter.api.Test
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Path
import org.junit.jupiter.api.Assertions.assertTrue

class VolBlockRegressionTest {

    private val volBlockSource: Path = Path.of("src/main/java/org/trp/shincolle/block/VolBlock.kt")
    private val modBlocksSource: Path = Path.of("src/main/java/org/trp/shincolle/init/ModBlocks.kt")
    private val modItemsSource: Path = Path.of("src/main/java/org/trp/shincolle/init/ModItems.kt")
    private val volBlockRecipe: Path =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolblock.json")
    private val volBlockRecipeAlt: Path =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolblock_2.json")
    private val volCoreRecipe: Path =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolcore.json")
    private val volCoreRecipeAlt: Path =
            Path.of("src/main/resources/data/shincolle/recipe/blockvolcore_3.json")

    @Test
    fun volBlockShouldStayRestoredForLegacyVolcanoCraftChain() {
        val blockSource = Files.readString(volBlockSource)
        val modBlocksSrc = Files.readString(modBlocksSource)
        val modItemsSrc = Files.readString(modItemsSource)
        val volBlockText = Files.readString(volBlockRecipe)
        val volBlockTextAlt = Files.readString(volBlockRecipeAlt)
        val volCoreText = Files.readString(volCoreRecipe)
        val volCoreTextAlt = Files.readString(volCoreRecipeAlt)

        assertTrue(modBlocksSrc.contains("VOL_BLOCK = BLOCKS.register(\"blockvolblock\"")) {
            "ModBlocks should keep registering the legacy volcano block"
        }
        assertTrue(modItemsSrc.contains("VOL_BLOCK = ITEMS.register(\"blockvolblock\"")) {
            "ModItems should keep exposing the volcano block as an item"
        }
        assertTrue(blockSource.contains("return true;")) {
            "VolBlock should stay valid as a beacon base like the legacy block"
        }
        assertTrue(volBlockText.contains("\"id\": \"shincolle:blockvolblock\"")) {
            "VolBlock recipe should stay present"
        }
        assertTrue(volBlockText.contains("\"item\": \"shincolle:grudge_block\"")) {
            "VolBlock recipe should keep using the legacy grudge block"
        }
        assertTrue(volBlockTextAlt.contains("\"id\": \"shincolle:blockvolblock\"")) {
            "VolBlock should keep the mirrored legacy recipe variant"
        }
        assertTrue(volCoreText.contains("\"item\": \"shincolle:blockvolblock\"")) {
            "VolCore recipe should keep consuming the restored volcano block"
        }
        assertTrue(volCoreTextAlt.contains("\"id\": \"shincolle:blockvolcore\"")) {
            "VolCore should keep the mirrored legacy recipe variant"
        }
    }
}
